clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -587,6 +588,35 @@ public void appendUncheckedWithOffset(long offset, LegacyRecord record) {
         }
     }
 
+    /**
+     * Append a record without doing offset/magic validation (this should only be used in testing).
+     *
+     * @param offset The offset of the record
+     * @param record The record to add
+     */
+    public void appendUncheckedWithOffset(long offset, SimpleRecord record) throws IOException {
```

Comment:
I'm a bit nervous about this method being in the public interface. Could we make it package private and then add a test utility method in the same package that calls it, exposing it only for tests?

Reply:
Oh, we already have a public `appendUncheckedWithOffset` for `LegacyRecord`.

Reply:
We could move these methods to a `TestMemoryRecordsBuilder` or something. I think risk of misuse is probably unlikely though.

Reply:
Makes sense. Since we already had one such method, I'm ok with keeping as is.